packages/flutter_tools/lib/src/ios/ios_emulators.dart

```diff
@@ -37,11 +37,23 @@ class IOSEmulator extends Emulator {
 
   @override
   Future<void> launch() async {
-    final RunResult launchResult =
-        await runAsync(<String>['open', '-a', getSimulatorPath()]);
-    if (launchResult.exitCode != 0) {
-      printError('$launchResult');
+    Future<void> launchSimulator(List<String> additionalArgs) async {
+      final List<String> args = <String>['open']
+          .followedBy(additionalArgs)
+          .followedBy(<String>['-a', getSimulatorPath()]);
+
+      final RunResult launchResult = await runAsync(args);
+      if (launchResult.exitCode != 0) {
+        printError('$launchResult');
+      }
     }
+
+    // First run with `-n` to force a device to boot if there isn't already one
+    await launchSimulator(<String>['-n']);
```

Comment:
Is there any possible harm in needlessly running this?  A few questions:

1. what happens if this errors?  Will things be even more confused if we try and run it again?
2. what does the simulator do if it's *never* been run?  That is, how can it choose a default device if you have never selected one? (I assume it just uses the last choice?  What if there was none?)

**EDIT:** looking for docs now...

Reply:
1. I think at worst we'd just show two errors. We could skip the second run if the first one fails though?

2. Er, not sure - I thought it opened something by default the first time I ran it. Is there a way we can clear its data to find out?

I did see that simctl seems to have options for listing and running specific devices, I did wonder whether there's anything useful in there (like allowing the user to select a specific device and then launch that) but it might get confusing if they think they can run multiple together (which I presume, they can not).

Reply:
Yeah, I looked at `simctl`.  A bit of a rabbit hole.  In fact that's how I missed this easier solution!  Down the road adding support for selecting devices and potentially other bits that `simctl` offers sounds good but IMO not pressing.

Skipping the second run on a failure sounds good, though maybe overly defensive.

Your call!